geo/src/algorithm/chaikins_smoothing.rs

```diff
@@ -0,0 +1,211 @@
+use std::ops::Mul;
+
+use num_traits::FromPrimitive;
+
+use crate::{CoordFloat, Coordinate, LineString, MultiLineString, MultiPolygon, Polygon};
+
+/// Smoothen `LineString`, `Polygon`, `MultiLineString` and `MultiPolygon` using Chaikins algorithm.
+///
+/// [Chaikins smoothing algorithm](http://www.idav.ucdavis.edu/education/CAGDNotes/Chaikins-Algorithm/Chaikins-Algorithm.html)
+///
+/// This implementation preserves the start and end vertices of an open linestring and
+/// smoothes the corner between start and end of a closed linestring.
+///
```

Comment:
Would it make sense to add a line explaining the effect of increasing `n_iterations`?

Reply:
Also a good point - I added that and reworded that comment.